### PR TITLE
[e2e] Fixes prepareProposal not to return oversized set of transactions

### LINKED ISSUE #1754: e2e test app may return oversized txs set by adding transaction with vote extensions.
TODO: fill in the details

### Setup

**CometBFT version** (use `cometbft version` or `git rev-parse --verify HEAD` if installed from source):

**Have you tried the latest version**: yes/no

**ABCI app** (name for built-in, URL for self-written if it's publicly available):

**Environment**:
- **OS** (e.g. from /etc/os-release):
- **Install tools**:
- **Others**:

**node command runtime flags**:

### PR DESCRIPTION
**Pull request by lasarojc**
Closes #1754 

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: test/e2e/app/app.go [hunk 1]
```diff
@@ -434,7 +435,8 @@ func (app *Application) PrepareProposal(
 		}
 		extCommitHex := hex.EncodeToString(extCommitBytes)
 		extTx := []byte(fmt.Sprintf("%s%d|%s", extTxPrefix, sum, extCommitHex))
-		extTxLen := int64(len(extTx))
+		extTxLen := int64(cmttypes.ComputeProtoSizeForTxs([]cmttypes.Tx{extTx}))
+
```
Comment (lasarojc) [inline test/e2e/app/app.go:439]:
```suggestion
```

File: test/e2e/app/app.go [hunk 2]
```diff
@@ -456,10 +457,10 @@ func (app *Application) PrepareProposal(
 			app.logger.Error("detected tx that should not come from the mempool", "tx", tx)
 			continue
 		}
-		if totalBytes+int64(len(tx)) > req.MaxTxBytes {
+		if totalBytes+cmttypes.ComputeProtoSizeForTxs([]cmttypes.Tx{tx}) > req.MaxTxBytes {
```
Comment (melekes) [inline test/e2e/app/app.go:460]: extract to variable to avoid double marshaling to proto?

Comment (lasarojc) [reply test/e2e/app/app.go]: Good point. Fixed.

File: .changelog/unreleased/bug-fixes/1754-no-oversized-blocks.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+- `[test]` Test app does not create oversized transaction set.
```
Comment (thanethomson) [inline .changelog/unreleased/bug-fixes/1754-no-oversized-blocks.md:1]: Do we know of any external users/developers who make use of the test app? If not, there's no need for a changelog entry.

**Review by thanethomson — APPROVED** [review]

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**